linera-chain/src/data_types.rs

```diff
@@ -600,15 +600,29 @@ impl<'a> LiteCertificate<'a> {
 }
 
 /// A certified statement from the committee.
-#[derive(Clone, Debug, Serialize)]
-#[cfg_attr(with_testing, derive(Eq, PartialEq))]
-pub struct Certificate {
-    /// The certified value.
-    pub value: HashedCertificateValue,
-    /// The round in which the value was certified.
-    pub round: Round,
-    /// Signatures on the value.
-    signatures: Vec<(ValidatorName, Signature)>,
+pub type Certificate = GenericCertificate<CertificateValue>;
+
+impl Serialize for Certificate {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        #[derive(Debug, Serialize)]
+        #[serde(rename = "Certificate")]
+        struct CertificateHelper {
+            value: CertificateValue,
+            round: Round,
+            signatures: Vec<(ValidatorName, Signature)>,
+        }
+
+        let helper = CertificateHelper {
+            value: self.inner().clone(),
+            round: self.round,
+            signatures: self.signatures().clone(),
+        };
+
+        helper.serialize(serializer)
+    }
```

Comment:
Actually, why do we need `Serialize` but not `Deserialize`?

Reply:
There exists a `Deserialize`, it's down below in this file (a custom `Deserializer` was already written for the `Certificate` type).